### PR TITLE
[scheduler] New way to add scheduler-extender

### PR DESCRIPTION
**Pull request by AleksZimin**
## Description

This PR transitions from using a dedicated scheduler with an extender for pods that utilize specific PVCs with `local.csi.storage.deckhouse.io` provisioner to a centralized approach provided by Deckhouse. Previously, a separate scheduler was required for each SDS module. With the introduction of the `KubeSchedulerWebhookConfiguration` resource in Deckhouse, we now configure the extender-scheduler directly within the default scheduler, eliminating the need for module-specific schedulers.

## Why do we need it, and what problem does it solve?

The previous method required maintaining a separate scheduler for each SDS module, leading to complexity and potential overhead. By adopting the `KubeSchedulerWebhookConfiguration`, we streamline the configuration process, reduce resource duplication, and centralize scheduler-extender configuration. This transition simplifies management and aligns with Deckhouse's best practices.

## What is the expected result?

- Centralized management of the scheduler-extender through the default scheduler.
- Removal of module-specific schedulers, reducing complexity and resource overhead.
- Improved alignment with Deckhouse's recommended practices for Kubernetes scheduling.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: hooks/generate_scheduler_extender_certs.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+def main():
+    hook = GenerateCertificateHook(
+        TlsSecret(
+            cn="sds-local-volume-scheduler-extender",
+            name="scheduler-extender-https-certs",
+            sansGenerator=default_sans([
+                "sds-local-volume-scheduler-extender",
+                f"sds-local-volume-scheduler-extender.{common.NAMESPACE}",
+                f"sds-local-volume-scheduler-extender.{common.NAMESPACE}.svc",
+                f"sds-local-volume-scheduler-extender.{common.NAMESPACE}.svc.cluster.local"]),
```
Comment (duckhawk) [inline hooks/generate_scheduler_extender_certs.py:33]: Don't need this, also, cluster can use suffix other than "cluster.local"

File: images/sds-local-volume-csi/pkg/utils/func.go [hunk 2]
```diff
@@ -276,20 +296,33 @@ func GetLVMThinPoolFreeSpace(lvg snc.LvmVolumeGroup, thinPoolName string) (thinP
...
+		err := kc.Update(ctx, llv)
 		if err == nil {
 			return nil
 		}
-		time.Sleep(KubernetesAPIRequestTimeout)
-	}
 
-	if err != nil {
-		return fmt.Errorf("after %d attempts of updating LVMLogicalVolume %s, last error: %w", KubernetesAPIRequestLimit, llv.Name, err)
+		if attempt < KubernetesAPIRequestLimit-1 {
```
Comment (duckhawk) [inline images/sds-local-volume-csi/pkg/utils/func.go:307]: Are we sure that we need multiple tries? It's not a k8s's way

File: templates/webhooks/webhook.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# {{- if and (not (eq "dev" .Values.global.deckhouseVersion)) (semverCompare "<1.64" .Values.global.deckhouseVersion) }}
```
Comment (duckhawk) [inline templates/webhooks/webhook.yaml:1]: Do we need this comment?

**Review by duckhawk — CHANGES_REQUESTED** [review]

File: hooks/generate_scheduler_extender_certs.py (hunk 1 above)
Comment (AleksZimin) [reply hooks/generate_scheduler_extender_certs.py]: Fixed

File: images/sds-local-volume-csi/pkg/utils/func.go (hunk 2 above)
Comment (AleksZimin) [reply images/sds-local-volume-csi/pkg/utils/func.go]: Fixed

File: templates/webhooks/webhook.yaml (hunk 3 above)
Comment (AleksZimin) [reply templates/webhooks/webhook.yaml]: Fixed

**Pull request merged**